infra/lambda/image-optimizer/lambda_function.py

```
@@ -0,0 +1,122 @@
+        new_height = int(original_height * ratio)
+        img = img.resize((MAX_WIDTH, new_height), Image.LANCZOS)
+        print(f"リサイズ: {original_width}x{original_height} -> {MAX_WIDTH}x{new_height}")
+    else:
+        print(f"リサイズ不要: {original_width}x{original_height}")
+
+    # WebP変換して保存
+    webp_buffer = io.BytesIO()
+    img.save(webp_buffer, format='WEBP', quality=WEBP_QUALITY, method=4)
+    webp_buffer.seek(0)
+
+    # 保存先キーを生成（元のファイル名.webp）
```

Ruff (0.15.5)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)